src/System.Net.Primitives/tests/FunctionalTests/CookieCollectionTest.cs

```diff
@@ -152,5 +153,73 @@ public static void SyncRoot_Get_Success()
             CookieCollection cc = new CookieCollection();
             Assert.Equal(cc, cc.SyncRoot);
         }
+
+        [Fact]
+        public static void CopyTo_Array_Success()
+        {
+            CookieCollection cc = CreateCookieCollection1();
+            Array c = new object[cc.Count];
+            cc.CopyTo(c, 0);
+            EnsureEqual(cc, c);
+        }
+
+        [Fact]
+        public static void CopyTo_CookieArray_Success()
+        {
+            CookieCollection cc = CreateCookieCollection1();
+            Cookie[] c = new Cookie[cc.Count];
+            cc.CopyTo(c, 0);
+            EnsureEqual(cc, c);
+        }
+
+        private static void EnsureEqual(CookieCollection cc, Array cookies)
+        {
+            Assert.Equal(cc.Count, cookies.Length);
+            for (int i = 0; i < cookies.Length; i++)
+            {
+                Assert.Equal(cc[i], cookies.GetValue(i));
+            }
+        }
+
+        [Fact]
+        public static void Enumerator_Index_Invalid()
+        {
+            CookieCollection cc = CreateCookieCollection1();
+            IEnumerator enumerator = cc.GetEnumerator();
+
+            Assert.Throws<InvalidOperationException>(() => enumerator.Current); // Index < 0
+
+            enumerator.MoveNext(); enumerator.MoveNext(); enumerator.MoveNext();
+            enumerator.MoveNext(); enumerator.MoveNext(); enumerator.MoveNext();
+
+            Assert.Throws<InvalidOperationException>(() => enumerator.Current); // Index >= count
+
+            enumerator.Reset();
+            Assert.Throws<InvalidOperationException>(() => enumerator.Current); // Index should be -1
+        }
+
+        [Fact]
+        public static void Enumerator_Version_Invalid()
+        {
+            CookieCollection cc = CreateCookieCollection1();
+            IEnumerator enumerator = cc.GetEnumerator();
+            enumerator.MoveNext();
+
+            cc.Add(new Cookie("name5", "value"));
+            Assert.Throws<InvalidOperationException>(() => enumerator.Current); // Enumerator out of sync
+            Assert.Throws<InvalidOperationException>(() => enumerator.MoveNext()); // Enumerator out of sync
+        }
+
+        [Fact]
+        public static void TestDump()
+        {
+#if DEBUG
+            CookieCollection cc = CreateCookieCollection1();
+            Type type = cc.GetType();
+
+            MethodInfo dumpMethod = type.GetTypeInfo().GetDeclaredMethod("Dump");
+            dumpMethod.Invoke(cc, null);
+#endif
```

Comment:
Yup you've got the aim of the test (it's also for code completion)

I'm also convinced that this might actually be dead code as it is not invoked anywhere from the project